lib/gemstash/http_client.rb

```diff
@@ -26,10 +26,11 @@ class HTTPClient
 
     DEFAULT_USER_AGENT = "Gemstash/#{Gemstash::VERSION}"
 
-    def self.for(upstream)
+    def self.for(upstream, timeout = 20)
       client = Faraday.new(upstream.to_s) do |config|
         config.use FaradayMiddleware::FollowRedirects
         config.adapter :net_http
+        config.options.timeout = timeout
       end
```

Comment:
Call it fetch_timeout for symmetry with the configuration variable and maybe consider making it a keyword argument?